ydb/library/yql/parser/pg_wrapper/arrow.cpp

```diff
@@ -190,12 +221,17 @@ TColumnConverter BuildPgNumericColumnConverter(const std::shared_ptr<arrow::Data
         };
     case arrow::Type::FLOAT:
         return [](const std::shared_ptr<arrow::Array>& value) {
-            return PgConvertNumeric<float>(value);
+             return PgConvertNumeric<float>(value);
         };
     case arrow::Type::DOUBLE:
         return [](const std::shared_ptr<arrow::Array>& value) {
             return PgConvertNumeric<double>(value);
         };
+    case arrow::Type::DECIMAL128:
+        return [originalType](const std::shared_ptr<arrow::Array>& value) {
+            auto decimal128Ptr = std::dynamic_pointer_cast<arrow::Decimal128Type>(originalType);
```

Comment:
performace issue: extract precision/scale before return then capture them in lambda